tests/test_slurmrestapi.py

```diff
@@ -0,0 +1,95 @@
+import os
+import time
+import json
+import urllib.request
+from http import HTTPStatus
+from http.client import HTTPResponse
+
+import pytest
+from testinfra.backend.base import CommandResult
+
+
+def get_slurm_token(token_output_content: str) -> str:
+    return token_output_content.strip().lstrip("SLURM_JWT=")
+
+
+def test_unauthorized(container_ip):
+    with pytest.raises(urllib.request.HTTPError, match="401: UNAUTHORIZED"):
+        urllib.request.urlopen(f"http://{container_ip}:6820/openapi")
+
+
+@pytest.mark.parametrize("username", ["slurm", "root"])
+def test_slurm_ping(host, container_ip, username):
+    token_output: CommandResult = host.run("scontrol token username=%s", username)
+    assert token_output.succeeded
+    jwt_token = get_slurm_token(token_output.stdout)
+
+    request_obj = urllib.request.Request(
+        f"http://{container_ip}:6820/slurm/v0.0.37/ping",
```

Comment:
Is hardcoding the API version going to be a problem in the future? Is there a way we could fetch the API version used in the test in conftest.py, for example, similar to the container ip?

Reply:
Good idea, I will fetch and get latest API path to request. Details can refer to https://slurm.schedmd.com/rest_api.html#Openapi